admission-webhook/webhook.go

```diff
@@ -393,6 +402,23 @@ func (webhook *webhook) mutateCreateRequest(ctx context.Context, pod *corev1.Pod
 	admissionResponse := &admissionV1.AdmissionResponse{Allowed: true}
 
 	if len(patches) != 0 {
+		// pods are GMSA related, we will need further check if we need to randomize the hostname
+		hostName := pod.Spec.Hostname
+		// Patch the hostname only if it is not empty
+		if webhook.config.EnableRandomHostName {
+			if hostName == "" {
+				hostName = generateUUID()
+				patches = append(patches, map[string]string{
+					"op":    "add",
+					"path":  "/spec/hostname",
+					"value": hostName,
+				})
+			} else {
+				// This will not happan in GMSA scenario, but added for completeness, just in case
+				return nil, &podAdmissionError{error: fmt.Errorf("can not set hostname in spec and also inidcating random hostname in ENV")}
```

Comment:
I think this will refuse to allow pod specs where the hostname has already been set. We shouldn't block users from setting there own hostname

Reply:
so I will just print out a message and let user stick to their own hostname in these cases